### PR TITLE
refactor MRGFY 346

### PR DESCRIPTION
**Pull request by CamClrt**

### CODE REVIEW

**Comment by CamClrt** [comment]
To be validated, it's necessary to check and merge this [PR](https://github.com/Mergifyio/mergify.io/pull/2044) before

**Review by sileht — DISMISSED** [review]
Looks good, but you need to rebase it, The new `train` queues mechanism have been merged yesterday and it changes some stuffs there.

File: mergify_engine/tests/functional/test_engine.py [hunk 1]
```diff
@@ -754,22 +754,6 @@ async def test_merge_smart_strict(self):
         await self.wait_for("pull_request", {"action": "synchronize"})
         await self.run_engine()
 
-        # with the older endpoint
```
Comment (sileht) [inline mergify_engine/tests/functional/test_engine.py:757]: We should keep that and just change the returned payload

Comment (sileht) [reply mergify_engine/tests/functional/test_engine.py:757]: You can also  complete the test in test_queue.py that query this API

File: mergify_engine/web/root.py [hunk 2]
```diff
@@ -275,22 +274,9 @@ async def queues_by_owner_id(
     ] = collections.defaultdict(dict)
     async for queue in redis_cache.scan_iter(match=f"merge-queue~{owner_id}~*"):
         _, _, repo_id, branch = queue.split("~")
-        async with github.aget_client(owner_id=owner_id) as client:
```
Comment (sileht) [inline mergify_engine/web/root.py:278]: Both endpoints must be exactly the same. And we should not duplicate the code.

**Review by sileht — CHANGES_REQUESTED** [review]

**Review by sileht — APPROVED** [review]

**Pull request merged**